pkg/deployment/depresource/upgrade_stateless_test.go

```diff
@@ -71,14 +140,35 @@ func TestUpgradeStateless(t *testing.T) {
 				API:          api.NewMock(mock.New202Response(mock.NewStringBody(""))),
 				DeploymentID: util.ValidClusterID,
 				Type:         "kibana",
+				RefID:        "main-kibana",
 			}},
 			want: new(models.DeploymentResourceUpgradeResponse),
 		},
+		{
+			name: "succeeds when RefID is not set",
+			args: args{params: UpgradeStatelessParams{
+				API: api.NewMock(
+					mock.New200Response(mock.NewStructBody(models.DeploymentGetResponse{
+						Healthy: ec.Bool(true),
+						ID:      ec.String("3531aaf988594efa87c1aabb7caed337"),
+						Resources: &models.DeploymentResources{
+							Elasticsearch: []*models.ElasticsearchResourceInfo{{
+								ID:    ec.String("3531aaf988594efa87c1aabb7caed337"),
+								RefID: ec.String("elasticsearch"),
+							}},
+						},
+					})),
+					mock.New200Response(mock.NewStringBody("")),
+				),
+				DeploymentID: util.ValidClusterID,
+				Type:         "elasticsearch",
+			}},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := UpgradeStateless(tt.args.params)
-			if !reflect.DeepEqual(err, tt.err) {
+			if tt.err != nil && err.Error() != tt.err.Error() {
```

Comment:
I think we should settle on which error assertions we want to follow. It's not a big deal, but we're not asserting the same thing, an error could be returned by the function anyway and we wouldn't catch it unless there's something in the `err` field of the test structure, which won't be if the test structure is not asserting for an error different than `nil`.

Reply:
I prefer using `!reflect.DeepEqual(err, tt.err)` as well :) In this case, unfortunately, when I added the last [test case](https://github.com/elastic/ecctl/pull/92/files#diff-2c76ec0b0bb37f47f4e839bf6f176a2dR148) I started getting the following error, so I had to change the assertion.

```
--- FAIL: TestUpgradeStateless/succeeds_when_RefID_is_not_set (0.00s)
    /Users/karen/go/src/github.com/elastic/ecctl/pkg/deployment/depresource/upgrade_stateless_test.go:172: UpgradeStateless() error = , wantErr <nil>
```

Reply:
huh that's weird

Reply:
DeepEquals probably fails because the tests are of different type...